packages/middleware-bucket-endpoint/src/configurations.ts

```diff
@@ -2,7 +2,8 @@ import { Provider, RegionInfoProvider } from "@aws-sdk/types";
 
 export interface BucketEndpointInputConfig {
   /**
-   * Whether the provided endpoint addresses an individual bucket.
+   * Whether to overwrite the request endpoint with the value of the `Bucket` in the request parameter. This option
+   * overwrites the custom endpoints too.
```

Comment:
Use the same documentation added originally in https://github.com/aws/aws-sdk-js-v3/pull/104/files#diff-9835c8f65fde8ebac11e2eee18105336c4ef55a3403977b12c8f77a81cf46c49R33-R42

```suggestion
   * Whether to use the bucket name as the endpoint for this request. The bucket
   * name must be a domain name with a CNAME record alias to an appropriate virtual
   * hosted-style S3 hostname, e.g. a bucket of `images.johnsmith.net` and a DNS
   * record of:
   *
   * ```
   * images.johnsmith.net CNAME 			images.johnsmith.net.s3.amazonaws.com.
   * ```
   *
   * @see https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#VirtualHostingCustomURLs
```